### DISCUSSION TITLE
Should json file I/O ever be done using std::ios_base::binary?

### DISCUSSION THREAD

**Discussion by WBIGitHub** [nlohmann/json#2827] [Q&A] [answered]
Previously, I've used boost::iostreams::filtering_ostream and istream to read and write compressed XML, and doing so required files to be opened with std::ios_base::binary.

Putting aside the issue of compression (not planning that for json), **_I assume that one should never open files to read/write plain-text json in binary mode?_**

Most of the json examples use the stream operators ("<<" and ">>"), which themselves do not support binary mode operations, so if opening in std::ios_base::binary mode was ever required, one would need to avoid using those operators.

_**Can anyone confirm that json file I/O is normally done in text mode?**_

Bump?

**Comment by nlohmann** [comment, marked as answer]
Reading JSON files as text or binary should not make a difference to this library.
